cloudwatch_logs_common/src/utils/log_file_manager.cpp

```diff
@@ -77,16 +67,19 @@ FileObject<LogCollection> LogFileManager::readBatch(
       log_data.push_front(input_event);
       data_tokens.push_back(new_data_token);
     }
-    else{
-      AWS_LOG_INFO(__func__, "Some logs were not batched since the time"
-        " difference was > 24 hours. Will try again in a separate batch./n"
-        "Logs read: %d, Logs batched: %d", batch_size, log_data.size()
-        );
-      break;
+    else if(file_manager_strategy_->isDeleteStaleData() && latestTime - curTime > TWO_WEEK_IN_SEC){
+      stale_data_.push_back(new_data_token);
```

Comment:
```suggestion
      {
         std::lock_guard<std::mutex> lock(active_delete_stale_data_mutex_);
         stale_data_.push_back(new_data_token);
      }
```

Lock with a small scope, as the `stale_data_` member needs to be protected and not the loop. 

* http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rconc-time
* https://stackoverflow.com/questions/54399719/including-stdlock-guard-in-extra-scope
* https://www.modernescpp.com/index.php/c-core-guidelines-concurrency-and-lock-free-programming